schema/gnosis_protocol/view_trades.sql

```diff
@@ -16,72 +16,75 @@ JOIN gnosis_protocol."BatchExchange_call_submitSolution" solution
 ),
 trades as (
 SELECT
-    solution."batchId" as batch_id,
-    trades.evt_block_time as block_time,
+	  -- id
+    floor(extract(epoch from solution.evt_block_time) / 300) - 1 AS batch_id, -- The event time tells us the batch. Between minute 0-4 is resolved batch N-1
+    trades."owner" AS trader_hex,    
     trades."orderId" as order_id,
-    trades."owner",
-    trades.evt_block_number as block_number,
-    trades.evt_index,
-    trades."buyToken" as buy_token_id,
-    trades."sellToken" as sell_token_id,
-    trades."executedBuyAmount" as executed_buy_amount,
-    trades."executedSellAmount" as executed_sell_amount,
-    trades.evt_tx_hash as tx_hash,
     RANK() OVER (
-        PARTITION BY solution."batchId", trades.owner, trades."orderId"
+        PARTITION BY solution.evt_block_time, trades.owner, trades."orderId"
         ORDER BY trades.evt_block_number, trades.evt_index
-    ) as trade_sub_id
-FROM gnosis_protocol."BatchExchange_evt_Trade" trades
-JOIN gnosis_protocol."BatchExchange_call_submitSolution" solution
-    ON solution.call_tx_hash=trades.evt_tx_hash
-    AND solution.call_success = true
-)
-SELECT
-    trades.batch_id,
-    trades.owner as trader_hex,
-    trades.order_id,
-    trades.trade_sub_id,
-    trades.block_time,
-    trades.block_number,
-    reverts.block_time as revert_time,
+    ) as trade_sub_id,
+    -- Event index
+    trades.evt_index AS evt_index_trades,
+    solution.evt_index AS evt_index_solution,
+    -- dates & block info
+    solution.evt_block_number,
+    solution.evt_block_time as block_time,
+    to_timestamp((floor(extract(epoch from solution.evt_block_time) / 300)) * 300) AS trade_date,
+    -- sell token
+    trades."sellToken" as sell_token_id,
     sell_token."token" as sell_token,
     sell_token."symbol" as sell_token_symbol,
     sell_token."decimals" as sell_token_decimals,
-    trades.executed_sell_amount as sell_amount_atoms,
-    trades.executed_sell_amount / 10^(sell_token.decimals) as sell_amount,
-    buy_token.token as buy_token,
+    -- sell amounts
+    trades."executedSellAmount" as sell_amount_atoms,
+    trades."executedSellAmount" / 10^(sell_token.decimals) as sell_amount,
+    -- buy token
+    trades."buyToken" as buy_token_id,
+    buy_token.token as buy_token,    
```

Comment:
If not to much trouble would be great to be consistent and use uppercase `AS` throughout the file